File: src/entity/entity.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 4,["7"] = 4,["8"] = 4,["9"] = 8,["10"] = 9,["11"] = 10,["12"] = 8});
local ____exports = {}
____exports.Entity = __TS__Class()
local Entity = ____exports.Entity
Entity.name = "Entity"
function Entity.prototype.____constructor(self, position, size)
    self.position = position:clone()
    self.size = position:clone()
end
return ____exports

File: src/entity/entity.ts
import { Vec2 } from "../vec2/vec2"
import { Obj } from "../object/object";
import { uuid } from "../utility/uuid";

let entity_map = new Map<string, Entity>();

export class Entity implements Obj {
  position: Vec2;
  size: Vec2;
  readonly uuid: string

  constructor(position: Vec2, size: Vec2) {
    this.position = position.clone();
    this.size = size.clone();
    this.uuid = uuid()

    print(this.uuid)
  }
}
